src/gt4py/next/program_processors/runners/dace_fieldview/transformations/loop_blocking.py

```diff
@@ -0,0 +1,607 @@
+        a direct connection to map entry.
+
+        The dependent nodes on the other side always have a direct edge to outer_entry.
+        As they are the set of nodes that are adjacent to outer_entry but are not
+        independent.
+
+        For the sake of arguments, all nodes, except the map entry and exit nodes,
+        that are inside the scope and are not classified as dependent or independent
+        are known as "inner nodes".
+
+        In case the function fails to compute the partition `None` is returned.
+
+        Args:
+            state: The state on which we operate.
+            sdfg: The SDFG in which we operate on.
+
+        Note:
+            - The function will only inspect the direct children of the map entry.
+            - Currently this function only considers the input Memlets and the
+                `used_symbol` properties of a Tasklet to determine if a Tasklet is dependent.
+        """
+        outer_entry: dace_nodes.MapEntry = self.outer_entry
+        blocking_parameter: str = self.blocking_parameter
+        independent_nodes: set[dace_nodes.Node] = set()  # `\mathcal{I}`
+
+        while True:
+            # Find all the nodes that we have to classify in this iteration.
+            #  - All nodes adjacent to `outer_entry`
+            #  - All nodes adjacent to independent nodes.
+            nodes_to_classify: set[dace_nodes.Node] = {
+                edge.dst for edge in state.out_edges(outer_entry)
+            }
+            for independent_node in independent_nodes:
+                nodes_to_classify.update({edge.dst for edge in state.out_edges(independent_node)})
+            nodes_to_classify.difference_update(independent_nodes)
+
+            # Now classify each node
+            found_new_independent_node = False
+            for node_to_classify in nodes_to_classify:
+                class_res = self.classify_node(
+                    node_to_classify=node_to_classify,
+                    outer_entry=outer_entry,
+                    blocking_parameter=blocking_parameter,
+                    independent_nodes=independent_nodes,
+                    state=state,
+                    sdfg=sdfg,
+                )
+
+                # Check if the partition exists.
+                if class_res is None:
+                    return None
+                if class_res is True:
+                    found_new_independent_node = True
+
+            # If we found a new independent node then we have to continue.
+            if not found_new_independent_node:
+                break
+
+        # After the independent set is computed compute the set of dependent nodes
+        #  as the set of all nodes adjacent to `outer_entry` that are not dependent.
+        dependent_nodes: set[dace_nodes.Node] = {
+            edge.dst for edge in state.out_edges(outer_entry) if edge.dst not in independent_nodes
+        }
+
+        return (independent_nodes, dependent_nodes)
+
+    def classify_node(
```

Comment:
If this function doesn't need to be a method because it doesn't use `self`, it should be declared as a module-level function outside of the class. Check here:  https://google.github.io/styleguide/pyguide.html#217-function-and-method-decorators
 > Never use staticmethod unless forced to in order to integrate with an API defined in an existing library. Write a module-level function instead.

Reply:
See my point above regarding `_rewire_map_scope()`. However, there is an inconsistency it should be private.